.github/workflows/docs.yml

```diff
@@ -0,0 +1,33 @@
+name: Docs
+
+on:
+  pull_request:
+    paths:
+      - 'docs/**'
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    name: docs
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - run: pip3 install -r docs/requirements.txt
+      - name: Build docs
+        run: |
+          WARNINGS_LOG=doc_warnings
+          SPELLING_LOG=_build/spelling
+          cd docs
+          sphinx-build -b spelling -n -q -w $WARNINGS_LOG -d _build/doctrees -D language=en_US . $SPELLING_LOG
```

Comment:
While we're at it, do we also want to use `-j auto` to speed things up?

```
  -j N              build in parallel with N processes where possible (special
                    value "auto" will set N to cpu-count)
```

Reply:
> While we're at it, do we also want to use `-j auto` to speed things up?

```console
$ make clean
rm -rf _build/*
$ time sphinx-build -b spelling -n -q -w doc_warnings -d _build/doctrees -D language=en_US . _build/spelling
real	1m41.217s
user	1m35.449s
sys	0m6.268s
$ make clean
rm -rf _build/*
$ time sphinx-build -b spelling -j auto -n -q -w doc_warnings -d _build/doctrees -D language=en_US . _build/spelling
real	1m15.066s
user	2m13.085s
sys	0m8.305s
```

Would we also be happy adding `-j auto -n` to `SPHINXOPTS` in `docs/Makefile` and `docs/make.bat`?
(Actually I see that `-n` is already in `ALLSPHINXOPTS` in `Makefile` but not `make.bat`...)

Reply:
> Any reason that `language=en_US` isn't just defined in `docs/conf.py`?
> (I guess translations of docs, but then passing something different on the command line will still override it.)

It's already in `docs/conf.py`, however I think it's good to keep it in the command line because it's more transparent for folks what language they should use (`en_US` vs. `en_GB`).

Reply:
> Would we also be happy adding -j auto -n to SPHINXOPTS in docs/Makefile and docs/make.bat?

It's not related with this change, feel-free to send a separate PR .

Reply:
Yeah, we can do it as a separate PR - we can also add caching of the doc trees to see if that speeds things up as well.